### PR TITLE
Restore TestTorchFunctionOverride

### LINKED ISSUE #74122: Override tests in test/test_overrides.py failing for some functions
### 🐛 Describe the bug

[`generate_tensor_like_override_tests(TestTorchFunctionOverride)`](https://github.com/pytorch/pytorch/blob/22a3a17ea0d61660b53f6642fe22529a7e4e2649/test/test_overrides.py#L693) in `test/test_overrides.py` was disabled since #64841. As a result, the associated tests have not been run since then. Re-enabling this in #74121 surfaced issues, particularly with 

- TestTorchFunctionOverride.test_torch_functional_histogramdd
- TestTorchFunctionOverride.test_Tensor_storage_type

These are skipped for now but should be fixed, opening this issue to track.

2022-03-11T20:23:14.4580034Z ERROR [0.001s]: test_torch_functional_histogramdd (__main__.TestTorchFunctionOverride)
2022-03-11T20:23:14.4580499Z ----------------------------------------------------------------------
2022-03-11T20:23:14.4580766Z Traceback (most recent call last):
2022-03-11T20:23:14.4580990Z   File "test_overrides.py", line 646, in test
2022-03-11T20:23:14.4581189Z     ret = func(*func_args, **kwargs)
2022-03-11T20:23:14.4581762Z   File "/opt/conda/lib/python3.7/site-packages/torch/functional.py", line 464, in histogramdd
2022-03-11T20:23:14.4582024Z     if bins and isinstance(bins[0], int):
2022-03-11T20:23:14.4582308Z TypeError: 'TensorLike' object is not subscriptable
2022-03-11T20:23:14.4582453Z 
2022-03-11T20:23:14.4582543Z ======================================================================
2022-03-11T20:23:14.4582785Z FAIL [0.001s]: test_Tensor_storage_type (__main__.TestTorchFunctionOverride)
2022-03-11T20:23:14.4583150Z ----------------------------------------------------------------------
2022-03-11T20:23:14.4583385Z Traceback (most recent call last):
2022-03-11T20:23:14.4583601Z   File "test_overrides.py", line 655, in test
2022-03-11T20:23:14.4583832Z     self.assertEqual(ret, -1)
2022-03-11T20:23:14.4584208Z   File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 2154, in assertEqual
2022-03-11T20:23:14.4584470Z     msg=msg,
2022-03-11T20:23:14.4584794Z   File "/opt/conda/lib/python3.7/site-packages/torch/testing/_comparison.py", line 1074, in assert_equal
2022-03-11T20:23:14.4585070Z     raise error_metas[0].to_error()
2022-03-11T20:23:14.4585349Z AssertionError: Object comparison failed: <class 'int'> != -1
2022-03-11T20:23:14.4585502Z 

### Versions

CI

cc @mruberry @hameerabbasi @rgommers @peterbell10

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74201
* __->__ #74202

Fixes #74122

This re-enables TestTorchFunctionOverride and fixes a bunch of test failures
that had crept in while it was disabled.

### CODE REVIEW

**Comment by ezyang** [comment]
Any word on when this will be finished? It would be helpful for these tests to be enabled for https://github.com/pytorch/pytorch/pull/75154

**Comment by peterbell10** [comment]
I've added a test skip for `histogramdd` which was the only thing blocking this so it's ready now.

File: test/test_overrides.py [hunk 1]
```diff
@@ -621,6 +622,9 @@ def instance_gen():
                     func_args.append(torch.float32)
                 elif t == 'c10::string_view':
                     func_args.append('')
+                elif t == 'SymInt':
+                    # TODO: generate actual SymbolicInt
+                    func_args.append(1)
```
Comment (ezyang) [inline test/test_overrides.py:627]: This is fine, SymInt will accept regular int too. (cc @Krovatkin, but no action necessary)

File: torch/_tensor.py [hunk 2]
```diff
@@ -862,6 +862,9 @@ def storage_type(self):
         Returns the type of the underlying storage.
 
         """
+        if has_torch_function_unary(self):
+            return handle_torch_function(Tensor.storage_type, (self,), self)
```
Comment (ezyang) [inline torch/_tensor.py:866]: nice

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**